src/SourceGenerators/CommonCode/CommonCode.csproj

```diff
@@ -0,0 +1,20 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <RootNamespace />
+    <EnableDefaultItems>false</EnableDefaultItems>
+    <CommonCodeVisible>true</CommonCodeVisible>
+  </PropertyGroup>
+
+  <!--
+    @rdeago 2022-05-21:
+    Without the IsPackable property, Visual Studio goes into an infinite loop of design-time builds
+    with IsPackable apparently alternating between true and false.
+    Observed on Visual Studio 20022 v17.2.1 and maybe related to the following issue:
+    https://github.com/dotnet/project-system/issues/1457
```

Comment:
Yuck! Another good reason not to do away with the `TargetFramework` property, I had issue with that in the past too...

Reply:
I suspect the problem stems from this code in `Directory.Build.targets`:

```xml
  <PropertyGroup Condition="'$(IsPackable)' == ''">
    <!-- The Sdks\NuGet.Build.Tasks.Pack\build\NuGet.Build.Tasks.Pack.targets unconditionally sets 
        PackageId=AssemblyName if no PackageId is provided, and then defaults IsPackable=true if 
        a PackageId is set (?!), meaning that by default everything is packable in Sdk-style 
        projects. 

        The Directory.Build.targets are imported after the user's project properties have been 
        read, and therefore gives us a chance to inspect if an explicit PackageId was provided, 
        before the NuGet SDK target is imported and defaults it. At this point, we can give 
        IsPackable a more sensible default, making it false if no PackageId was provided at this 
        point. -->
    <IsPackable Condition="'$(PackageId)' == ''">false</IsPackable>
    <IsPackable Condition="'$(PackageId)' != ''">true</IsPackable>
  </PropertyGroup>
```

Maybe something has changed in the SDK so that now `IsPackable` is checked (and defaulted to `true`) before `Directory.Build.targets` is included.

I won't modify `Directory.Build.targets` because it is synced with `devlooped/oss`. Further investigation (which unfortunately I have no time to carry out at the moment) may reveal an issue in `devlooped/oss`. For now, setting `IsPackable` in the project file looks like a viable workaround.

Reply:
Trying this in a clean project and I can't repro. So I'm guessing it might be something else...